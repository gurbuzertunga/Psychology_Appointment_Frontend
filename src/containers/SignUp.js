import { connect } from 'react-redux';
import { useHistory } from 'react-router-dom';
import PropTypes from 'prop-types';
import SignUpForm from '../components/signUpForm';
import signUpRequest from '../requests/signUpRequest';
import {
  createToken,
  createUser,
  closeModal,
  openModal,
} from '../actions/index';

const SignUp = (
  {
    createUser,
    createToken,
    closeModal,
    openModal,
  },
) => {
  const history = useHistory();
  const handleSubmit = data => {
    const { name, email } = data;
    createUser({ name, email });
    signUpRequest(createToken, data);
    setTimeout(() => {
      history.push('/consultancies');
    }, 2000);
    closeModal();
  };
  return (
    <div className="flex flex-col">
      <h1 className="text-center my-4 text-white lg:text-2xl text-4xl font-extrabold">Sign Up</h1>
      <SignUpForm handleClick={handleSubmit} />
      <div className="font-light text-center lg:text-base text-2xl cursor-pointer" onClick={() => openModal('login')} onKeyDown={openModal} role="presentation">Already have an Account?</div>
    </div>
  );
};
SignUp.propTypes = {
  createToken: PropTypes.func,
  createUser: PropTypes.func,
  closeModal: PropTypes.func,
  openModal: PropTypes.func,
};

SignUp.defaultProps = {
  createToken,
  createUser,
  closeModal,
  openModal,
};
export default connect(null, {
  createToken,
  createUser,
  closeModal,
  openModal,
})(SignUp);
