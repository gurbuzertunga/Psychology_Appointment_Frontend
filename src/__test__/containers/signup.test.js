import React from 'react';

import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import SignUp from '../../containers/SignUp';

configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({});
const setup = () => shallow(
  <Provider store={store}>
    <SignUp />
  </Provider>,
);

describe('SignUp Component', () => {
  let component;
  beforeEach(() => {
    component = setup();
  });
  test('renders the Consultancies List Component', () => {
    const page = component.find('.consultancyList');
    expect(page).toMatchSnapshot();
  });
});
