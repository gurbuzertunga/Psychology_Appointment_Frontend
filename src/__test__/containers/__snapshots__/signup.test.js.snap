// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp Component renders the Consultancies List Component 1`] = `ShallowWrapper {}`;
